test/Tester/DependencyInjectionGrainTests.cs

```diff
@@ -39,7 +41,49 @@ protected override TestCluster CreateTestCluster()
 
             // the injected service will return the same value only if it's the same instance
             Assert.Equal(
-                await grain1.GetStringValue(), 
+                await grain1.GetStringValue(),
+                await grain2.GetStringValue());
+        }
+
+        [Fact, TestCategory("BVT"), TestCategory("Functional")]
+        public async Task CannotGetExplictlyRegisteredGrain()
+        {
+            ISimpleDIGrain grain = GrainFactory.GetGrain<ISimpleDIGrain>(GetRandomGrainId(), grainClassNamePrefix: "UnitTests.Grains.ExplicitlyRegistered");
+            var exception = await Assert.ThrowsAsync<OrleansException>(() => grain.GetTicksFromService());
+            Assert.Contains("Error creating activation for", exception.Message);
+            Assert.Contains(nameof(ExplicitlyRegisteredSimpleDIGrain), exception.Message);
+        }
+    }
+
+    public class DelegateDependencyInjectionGrainTests : OrleansTestingBase, IClassFixture<DelegateDependencyInjectionGrainTests.Fixture>
+    {
+        private class Fixture : BaseTestClusterFixture
+        {
+            protected override TestCluster CreateTestCluster()
+            {
+                return new TestCluster(new TestClusterOptions(1))
+                {
+                    AppDomainSiloCreatorType = typeof(SiloCreator)
```

Comment:
I understand this change, just a heads up that this will probably be another thing we'll have to work around if we want a TestCluster that works in CoreCLR (where AppDomains are not available, and we'll probably have to resort to starting processes and IPC, unless we solve the issue with statics, which is very unlikely).
Not necessarily an action item for this PR, but just a heads up in case you thought of other approaches that are less reliant on AppDomains (or maybe just removing AppDomain suffix from the property name, as alternate implementations might use that creator type differently).